app/utils/extract_thumbnail.py

```
@@ -22,4 +27,28 @@ def extract_thumbnail(html_content: str) -> str:
         return og_image['content']
     else:
         return "basetumbnail.jpg"
+
+async def get_thumbnail_url(url: str) -> Optional[str]:
+    """URL에서 썸네일 이미지 URL을 추출합니다."""
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, timeout=aiohttp.ClientTimeout(total=10)) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)